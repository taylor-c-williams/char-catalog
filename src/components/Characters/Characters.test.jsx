import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Characters from './Characters';
import App from '../../App';

const characters = [
  {
    id: 1,
    image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
    name: 'Rick Sanchez',
    species: 'human',
    status: 'alive',
  },
  {
    id: 2,
    image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
    name: 'Morty Smith',
    species: 'Human',
    status: 'Alive',
  },
];
describe('Characters', () => {
  it('renders all characters', async () => {
    const container = render(
      <MemoryRouter initialEntries={['/characters']}>
        <Characters characters={characters} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
