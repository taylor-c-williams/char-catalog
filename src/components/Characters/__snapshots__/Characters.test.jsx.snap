// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters renders all characters 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        aria-label="characters"
        class="list"
      >
        <li
          class="item"
        >
          <a
            href="/characters/1"
          >
            <figure>
              <img
                alt="Rick Sanchez"
                class="image"
                src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
              />
              <figcaption
                class="linkText"
              >
                Rick Sanchez
              </figcaption>
            </figure>
          </a>
        </li>
        <li
          class="item"
        >
          <a
            href="/characters/2"
          >
            <figure>
              <img
                alt="Morty Smith"
                class="image"
                src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
              />
              <figcaption
                class="linkText"
              >
                Morty Smith
              </figcaption>
            </figure>
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      aria-label="characters"
      class="list"
    >
      <li
        class="item"
      >
        <a
          href="/characters/1"
        >
          <figure>
            <img
              alt="Rick Sanchez"
              class="image"
              src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            />
            <figcaption
              class="linkText"
            >
              Rick Sanchez
            </figcaption>
          </figure>
        </a>
      </li>
      <li
        class="item"
      >
        <a
          href="/characters/2"
        >
          <figure>
            <img
              alt="Morty Smith"
              class="image"
              src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
            />
            <figcaption
              class="linkText"
            >
              Morty Smith
            </figcaption>
          </figure>
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
